openwhisk/openwhisk.yaml

```diff
@@ -0,0 +1,100 @@
+              description: name of the function in the "__OW_ACTION_CODE" to call as the action handler
+              default: "main"
+            - name: OW_ACTION_BINARY
+              description: flag to indicate zip function, for zip actions, "__OW_ACTION_CODE" must be base64 encoded string
+              default: "false"
+            - name: OW_HTTP_METHODS
+              description: list of HTTP methods, any combination of [GET, POST, PUT, and DELETE], default is [POST]
+              default: "[POST]"
+            - name: OW_ACTION_RAW
+              description: flag to indicate raw HTTP handling, interpret and process an incoming HTTP body directly
+              default: "false"
+            - name: OW_PROJECT_URL
+              description: Location to local or remote file storage or public or private GitHub repo from where JavaScript source code needs to be evaluated
+              default: ""
+    outputs:
+        resources:
+            - name: runtime-image
+              type: image
+    steps:
+        - name: add-ow-env-to-dockerfile
+          image: "gcr.io/kaniko-project/executor:debug"
+          command:
+            - /busybox/sh
+          args:
+            - -c
+            - |
+              if [ -z ${inputs.params.OW_PROJECT_URL} ]; then
+                OW_ACTION_CODE="${inputs.params.OW_ACTION_CODE}"
+              else
+                TEMPDIR="knative-"$((1 + RANDOM % 100))
+                TEMPFILE=`basename "${inputs.params.OW_PROJECT_URL}"`
+                mkdir $TEMPDIR
+                cd $TEMPDIR
+                wget -O $TEMPFILE "${inputs.params.OW_PROJECT_URL}"
+                if [ "${inputs.params.OW_ACTION_BINARY}" = true ]; then
+                    OW_ACTION_CODE=`base64 $TEMPFILE | tr -d '[:space:]'`
+                else
+                    OW_ACTION_CODE=`cat $TEMPFILE`
+                fi
+                cd ..
+              fi
+              cat <<EOF >> ${inputs.params.DOCKERFILE}
+                ENV __OW_RUNTIME_DEBUG "${inputs.params.OW_RUNTIME_DEBUG}"
+                ENV __OW_RUNTIME_PLATFORM "${inputs.params.OW_RUNTIME_PLATFORM}"
+                ENV __OW_ACTION_NAME "${inputs.params.OW_ACTION_NAME}"
+                ENV __OW_ACTION_CODE "${OW_ACTION_CODE}"
+                ENV __OW_ACTION_MAIN "${inputs.params.OW_ACTION_MAIN}"
+                ENV __OW_ACTION_BINARY "${inputs.params.OW_ACTION_BINARY}"
+                ENV __OW_HTTP_METHODS "${inputs.params.OW_HTTP_METHODS}"
+                ENV __OW_ACTION_RAW "${inputs.params.OW_ACTION_RAW}"
+                ENV __OW_PROJECT_URL "${inputs.params.OW_PROJECT_URL}"
+              EOF
+        - name: update-dockerfile-for-tekton
+          image: "gcr.io/kaniko-project/executor:debug"
+          command:
+            - sed
+          args:
+            - -i
+            - -e
+            - 's/COPY ./COPY .\/runtime-git/g'
+            - ${inputs.params.DOCKERFILE}
+        - name: build-openwhisk-nodejs-runtime
```

Comment:
I wonder if the Task should stop at building the docker file and sharing it along with the action code on a storage.
There are already Tasks defined in the catalog the take care of building and pushing a docker image to a registry, so I think it would be nice to use this Task in a pipeline that uses a second Task for build purposes.

The other side of the coin is that this Task as it is defined here can be used independently, so it's quite handy and it does not require any storage to share data between Tasks. Perhaps in future there could be two flavours of this, the one defined here an another one with a storage output resource to be used in combination with a build task.

Reply:
like the way its independent, as its handy like you mentioned 👍